packages/flame/pubspec.yaml

```diff
@@ -14,6 +14,7 @@ dependencies:
   collection: ^1.15.0
   ordered_set: ^5.0.0
   vector_math: ^2.1.1
+  bitmap: ^0.1.2
```

Comment:
I don't think we'd want to pull in another dependency for this.

Reply:
I was already working on removing that as a dependency so that is reflected in the most recent commit.

I am having trouble with the caching however. It seems that after running `_images.add('$path$_generatedAtlasKeySuffix', atlas)` the `atlas` image is disposed. I've also tried setting `atlas` to the result of `_images.fromCache` after adding it to the cache but this still results in `atlas` being disposed.

Reply:
I figured out the caching - the problem was because all the layers and images of the `SpriteBatch` are being loaded asynchronously (with `Future.wait`). The image generation ends up being called multiple times before it's ever added to the cache the first time.

I added a variation of `add` and `fromCache` to `Images` that returns `Future<Image>` and with that I am able to get caching working.